src/app/drive/components/FileViewer/utils/fileViewerWrapperUtils.ts

```diff
@@ -0,0 +1,222 @@
+  currentFile,
+  user,
+  onClose,
+  onShowStopSharingDialog,
+  driveItemActions,
+  isCurrentUserViewer,
+}: {
+  currentFile: PreviewFileItem;
+  user: UserSettings | null;
+  onClose: () => void;
+  onShowStopSharingDialog: (() => void) | undefined;
+  driveItemActions: DriveItemActions;
+  isCurrentUserViewer: () => boolean;
+}): TopBarActionsMenu => {
+  const path = getAppConfig().views.find((view) => view.path === location.pathname);
+  const pathId = path?.id as PathProps;
+  const isRecentsView = pathId === 'recents';
+  const isSharedView = pathId === 'shared';
+  const isTrashView = pathId === 'trash';
+
+  const isSharedItem = (currentFile.sharings && currentFile.sharings?.length > 0) ?? false;
+  const isOwner = currentFile.credentials?.user === user?.email;
+
+  const {
+    onDownloadItemButtonClicked,
+    onMoveItemButtonClicked,
+    onCopyLinkButtonClicked,
+    onShowDetailsButtonClicked,
+    onLinkSettingsButtonClicked,
+    onMoveToTrashButtonClicked,
+    onRenameItemButtonClicked,
+    onRestoreItemButtonClicked,
+    onDeletePermanentlyButtonClicked,
+  } = driveItemActions;
+
+  const driveActionsMenu = (): ListItemMenu<DriveItemData> => {
+    if (isSharedItem) {
+      return contextMenuDriveItemShared({
+        copyLink: onCopyLinkButtonClicked,
+        openShareAccessSettings: onLinkSettingsButtonClicked,
+        deleteLink: () => ({}),
+        showDetails: onShowDetailsButtonClicked,
+        renameItem: onRenameItemButtonClicked,
+        moveItem: onMoveItemButtonClicked,
+        downloadItem: onDownloadItemButtonClicked,
+        moveToTrash: () => {
+          onMoveToTrashButtonClicked();
+          onClose();
+        },
+      });
+    } else {
+      return contextMenuDriveNotSharedLink({
+        shareLink: onLinkSettingsButtonClicked,
+        getLink: onCopyLinkButtonClicked,
+        renameItem: onRenameItemButtonClicked,
+        showDetails: onShowDetailsButtonClicked,
+        moveItem: onMoveItemButtonClicked,
+        downloadItem: onDownloadItemButtonClicked,
+        moveToTrash: () => {
+          onMoveToTrashButtonClicked();
+          onClose();
+        },
+      });
+    }
+  };
+
+  const recentsActionsMenu = (): ListItemMenu<DriveItemData> => {
+    return contextMenuDriveNotSharedLink({
+      shareLink: onLinkSettingsButtonClicked,
+      getLink: onCopyLinkButtonClicked,
+      renameItem: onRenameItemButtonClicked,
+      moveItem: onMoveItemButtonClicked,
+      showDetails: onShowDetailsButtonClicked,
+      downloadItem: onDownloadItemButtonClicked,
+      moveToTrash: () => {
+        onMoveToTrashButtonClicked();
+        onClose();
+      },
+    });
+  };
+
+  const sharedActionsMenu = (): ListItemMenu<AdvancedSharedItem> => {
+    return contextMenuDriveItemSharedAFS({
+      openShareAccessSettings: isOwner ? onLinkSettingsButtonClicked : undefined,
+      copyLink: onCopyLinkButtonClicked,
+      deleteLink: () => undefined,
```

Comment:
this looks like code smell, why is this a NOOP?

Reply:
Because we don't use delete link anymore since we added the new Shared. But yeah, I think we have to remove this option.